src/id.rs

```diff
@@ -679,6 +681,67 @@ impl NodeId {
         Ok(())
     }
 
+    /// Creates and appends a new node (from its associated data) as the last child.
+    /// This method is a fast path for the common case of appending a new node. It is quicker than [`append`].
+    ///
+    /// # Panics
+    ///
+    /// Panics if the arena already has `usize::max_value()` nodes.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use indextree::Arena;
+    /// let mut arena = Arena::new();
+    /// let n1 = arena.new_node("1");
+    /// let n1_1 = n1.append_value("1_1", &mut arena);
+    /// let n1_1_1 = n1_1.append_value("1_1_1", &mut arena);
+    /// let n1_1_2 = n1_1.append_value("1_1_2", &mut arena);
+    ///
+    /// // arena
+    /// // `-- 1
+    /// //     |-- 1_1
+    /// //  1_1_1 --|-- 1_1_2
+    ///
+    /// let mut iter = n1.descendants(&arena);
+    /// assert_eq!(iter.next(), Some(n1));
+    /// assert_eq!(iter.next(), Some(n1_1));
+    /// assert_eq!(iter.next(), Some(n1_1_1));
+    /// assert_eq!(iter.next(), Some(n1_1_2));
+    /// assert_eq!(iter.next(), None);
+    /// ```
+    /// [`append`]: struct.NodeId.html#method.append
+    pub fn append_value<T>(self, value: T, arena: &mut Arena<T>) -> NodeId {
+        let new_child = arena.new_node(value);
+        self.unchecked_append_new_node(new_child, arena);
+
+        new_child
+    }
+
+    /// Appends a new child to this node, after existing children.
+    /// This method is a fast path for the common case of appending a new node.
+    /// `new_child` requirements:
+    /// 1. Must be detached. No parents or siblings.
+    /// 2. Has not been [`remove`]d.
+    /// 3. `append_new_node()` was not called on itself
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use indextree::Arena;
+    /// let mut arena = Arena::new();
+    /// let n1 = arena.new_node("1");
+    /// assert!(n1.checked_append(n1, &mut arena).is_err());
+    ///
+    /// let n1_1 = arena.new_node("1_1");
+    /// assert!(n1.checked_append(n1_1, &mut arena).is_ok());
+    /// ```
+    ///
+    /// [`remove`]: struct.NodeId.html#method.remove
+    fn unchecked_append_new_node<T>(self, new_child: NodeId, arena: &mut Arena<T>) {
```

Comment:
nits: `unchecked` at beginning or end? need consistency

Reply:
end of the function, by Rust conventions.